experiments/modern/src/maki-interpreter/virtualMachine.js

```diff
@@ -134,16 +142,28 @@ async function interpret(start, program, { logger = null }) {
         stack.push(obj[methodName](...methodArgs));
         break;
       }
+      // callGlobal
+      case 25: {
+        const offset = command.arguments[0].offset;
+        const nextCommandIndex = offsetToCommand[offset];
+        const value = await interpret(nextCommandIndex, program, stack, { logger });
+        stack.push(value);
+        break;
+      }
       // return
       case 33: {
-        const variable = stack.pop();
-        return variable.getValue();
+        const a = stack.pop();
+        const aValue = a instanceof Variable ? a.getValue() : a;
+        return aValue;
       }
       // mov
       case 48: {
         const a = stack.pop();
         const b = stack.pop();
-        const aValue = a instanceof Variable ? a.getValue() : a;
+        let aValue = a instanceof Variable ? a.getValue() : a;
+        if (b.type === "INT" && !Number.isInteger(aValue)) {
+          aValue = Math.floor(aValue);
```

Comment:
I wonder if it would be faster to just always hit it with `floor` instead of checking `isInteger`

Reply:
Yeah, that makes sense to me 👍